### DISCUSSION TITLE
Changing the spiral coaster lift hill so that it is a block section.

### DISCUSSION THREAD

**Discussion by Joris-Driessen** [OpenRCT2/OpenRCT2#21562] [Feature requests]
A spiral chain lift currently does not work like a block section. This means that when a player uses the spiral chain section up, they would get one less block section, and one less train than the other coasters. A current workaround is to place a block brake at the top of the lift hill. Changing this, so that it is a block section would make it consistent with all the other coasters.
![2024-03-09 18_25_54-OpenRCT2](https://github.com/OpenRCT2/OpenRCT2/assets/63918462/c07d081f-272c-41e8-b164-c8067a4135f7)
![2024-03-09 18_26_37-OpenRCT2](https://github.com/OpenRCT2/OpenRCT2/assets/63918462/64758b15-1766-44a1-9466-dc2f05694f6f)

**Comment by ZeeMaji** [comment]
Changing this would potentially break compatibility with old designs, wouldn't it? Especially those that use a block section at the top already as the game blocks you from opening rides with block pieces that are right next to eachother.

**Reply by X123M3-256** [reply]
Not if you're smart about the implementation.

Consider that the reason the spiral lift is not normally a block section is because the lift is made up of the same piece repeated ... unlike normal chain lifts where it is the gentle to flat piece that registers as a block. So, I think to make this work, you'd have to add a new "sprial lift with block" piece and make it so that the last piece in the lift is automatically changed to the block version. This is similar to the station pieces, where the end piece with the traffic lights is actually a distinct track piece from the standard station piece.

This should solve issues with backward compatibility, as earlier rides would not use the new piece, they would only get it if you went into construction mode and edit the lift. However, if you did do that, you would then have no way to put it back how it was before.

**Reply by ZeeMaji** [reply]
Makes sense, although wouldn't this still affect old td6 designs due to how they're placed? Afaik when you place a td6 it effectively does a bunch of building actions all at once to create the whole design, wouldn't that also trigger placing the new "spiral lift with block" piece during that process? Or would this directly place the non-block version of the lift regardless of whether or not it's the last piece?

**Reply by X123M3-256** [reply]
That I don't know, not without looking at the code. I haven't checked how the station pieces are handled - I don't know if the game determines whether to place an end piece or regular piece before the game action, or if this takes place inside the action. However, game actions didn't exist in the original game - so I'm pretty sure that the TD6 file will store the station end piece as a different track piece. So, it ought not to be too difficult to make it so the TD6 loader would respect what's in the file.

I can't be bothered to look at the code but I might be able to test this by seeing if I can save and then load a design which features a station without an end piece, without it re-adding the end piece.
